datafusion/core/tests/simplification.rs

```diff
@@ -658,3 +658,18 @@ fn test_simplify_concat() {
     let expected = concat(vec![col("c0"), lit("hello rust"), col("c1")]);
     test_simplify(expr, expected)
 }
+
+#[test]
+fn test_simplify_iterations() {
```

Comment:
does this test fail without the change? 

Maybe we could add the test from the ticket too: https://github.com/apache/datafusion/issues/1160#issuecomment-952905624

@devinjdangelo 's deeply nested expressions usecase would also be interesting to test: https://github.com/apache/datafusion/issues/1160#issuecomment-1986906627

Reply:
> does this test fail without the change?

This was the example given in the original issue, though slightly modified to make it more test friendly. I thought it was failing without the change, though checking again now that appears to not be the case. I likely forgot to retest once I went down the `log` simplifier bug  rabbit hole.

> Maybe we could add the test from the ticket too: https://github.com/apache/datafusion/issues/1160#issuecomment-952905624

@devinjdangelo 's deeply nested expressions usecase would also be interesting to test: https://github.com/apache/datafusion/issues/1160#issuecomment-1986906627

Good idea. I can add those.

I can also add the `simplify_inner` suggestion and then explicitly test the number of iterations used.